src/f5_tts/model/utils.py

```diff
@@ -69,8 +69,10 @@ def mask_from_frac_lengths(seq_len: int["b"], frac_lengths: float["b"]):  # noqa
 def maybe_masked_mean(t: float["b n d"], mask: bool["b n"] = None) -> float["b d"]:  # noqa: F722
     if not exists(mask):
         return t.mean(dim=1)
-
-    t = torch.where(mask[:, :, None], t, torch.tensor(0.0, device=t.device))
+    
+    mask = mask.to(device=t.device, dtype=torch.bool)
+    mask = mask.unsqueeze(-1) 
+    t = torch.where(mask, t, torch.zeros_like(t))
```

Comment:
no sure, if critical to modify this part

Reply:
Changed back, proved non-critical